go/opentelemetry/collector/client/client.go

```diff
@@ -0,0 +1,163 @@
+//
+// example code to test lightstep/opentelemetry-exporter-go
+//
+// NOTE: Requires that you run a Collector instance
+// (see collector/vanilla/readme.md for details on how to run the
+// Collector locally)
+//
+// usage:
+//   go run client.go
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"time"
+
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.10.0"
+	"go.opentelemetry.io/otel/trace"
+)
+
+var (
+	tracer         trace.Tracer
+	serviceName    = os.Getenv("LS_SERVICE_NAME")
+	serviceVersion = os.Getenv("LS_SERVICE_VERSION")
+	endpoint       = os.Getenv("LS_SATELLITE_URL")
+	lsEnvironment  = os.Getenv("LS_ENVIRONMENT")
+	targetURL      = os.Getenv("DESTINATION_URL")
+)
+
+func newExporter(ctx context.Context) (*otlptrace.Exporter, error) {
+
+	if len(endpoint) == 0 {
+		endpoint = "localhost:4317"
+		log.Printf("Using default LS endpoint %s", endpoint)
+	}
+
+	exporter, err :=
+		otlptracegrpc.New(ctx,
+			// WithInsecure lets us use http instead of https.
+			// This is just for local development.
+			otlptracegrpc.WithInsecure(),
+			otlptracegrpc.WithEndpoint(endpoint),
+		)
+
+	return exporter, err
+}
+
+func newTraceProvider(exp *otlptrace.Exporter) *sdktrace.TracerProvider {
+
+	if len(serviceName) == 0 {
+		serviceName = "test-go-client-collector"
+		log.Printf("Using default service name %s", serviceName)
+	}
+
+	if len(serviceVersion) == 0 {
+		serviceVersion = "0.1.0"
+		log.Printf("Using default service version %s", serviceVersion)
+	}
+
+	if len(lsEnvironment) == 0 {
+		lsEnvironment = "dev"
+		log.Printf("Using default environment %s", lsEnvironment)
+	}
```

Comment:
this variable isn't used anywhere

Reply:
Ooops...should've been referenced on line 92 and I instead hardcoded the value.